osu.Game.Rulesets.Mania/Scoring/ManiaScoreProcessor.cs

```diff
@@ -19,15 +160,132 @@ public ManiaScoreProcessor(HitRenderer<ManiaHitObject, ManiaJudgement> hitRender
         {
         }
 
+        protected override void ComputeTargets(Beatmap<ManiaHitObject> beatmap)
+        {
+            BeatmapDifficulty difficulty = beatmap.BeatmapInfo.Difficulty;
+            hpMultiplier = BeatmapDifficulty.DifficultyRange(difficulty.DrainRate, hp_multiplier_min, hp_multiplier_mid, hp_multiplier_max);
+            hpMissMultiplier = BeatmapDifficulty.DifficultyRange(difficulty.DrainRate, hp_multiplier_miss_min, hp_multiplier_miss_mid, hp_multiplier_miss_max);
+
+            while (true)
+            {
+                foreach (var obj in beatmap.HitObjects)
+                {
+                    if (obj is Note)
+                    {
+                        AddJudgement(new ManiaJudgement
+                        {
+                            Result = HitResult.Hit,
+                            ManiaResult = ManiaHitResult.Perfect
+                        });
+                    }
+                    else if (obj is HoldNote)
+                    {
+                        // Head
+                        AddJudgement(new ManiaJudgement
+                        {
+                            Result = HitResult.Hit,
+                            ManiaResult = ManiaJudgement.MAX_HIT_RESULT
+                        });
+
+                        // Ticks
+                        for (int i = 0; i < ((HoldNote)obj).TotalTicks; i++)

```

Comment:
this count will evaluate every loop. i'd remove `TotalTicks` and just figure it out locally here.